src/core/grid_based_algorithms/lbgpu_cuda.cu

```diff
@@ -818,26 +824,24 @@ __device__ void thermalize_modes(float *mode, unsigned int index,
   c = (mode[0 + 0 * LBQ] +
        para->rho[0] * para->agrid * para->agrid * para->agrid) /
       Rho_tot;
-  random_wrapper(rn);
   for (int ii = 0; ii < LB_COMPONENTS; ++ii) {
     mode[1 + ii * LBQ] +=
         sqrtf(c * (1 - c) * Rho_tot *
               (para->mu[ii] * (2.0f / 3.0f) *
                (1.0f - (para->gamma_mobility[0] * para->gamma_mobility[0])))) *
-        (2 * ii - 1) * rn->randomnr[0];
+        (2 * ii - 1) * random_wrapper_philox(&n_a.philox_state[index]);
```

Comment:
You don't need to store a per-cell state object. That's just a waste of memory. If you do as I suggested above, you can use the cell coordinate and the mode number as the counter and the seed and the time step as the key.